File: todo.py
# version - 1.0.4

def add(li):
    task = input("Write task to add: ")
    li.append(task)
    print("<< ", li, " >>")
    make_list()

def remove(li):
    print('\n')
    erase = int(input("\nEnter task number to remove from the list: "))
    if erase < len(li):
        li.pop(erase)
        print("<< ", li, " >>")
    else:
        print("Invalid task number.")
    make_list()

def update(li):
    print('\n')
    task_no = int(input("\nEnter task number to update: "))
    if task_no < len(li):
        new_task = input("Enter new task: ")
        li[task_no] = new_task
        print("<< Updated list: ", li, " >>")
    else:
        print("Invalid task number.")
    make_list()

def make_list():
    choice = int(input("\n1 to add, 2 to remove, 3 to update, 0 to exit: "))
    if choice == 1:
        add(todo)
    elif choice == 2:
        remove(todo)
    elif choice == 3:
        update(todo)
    elif choice == 0:
        print("\nExiting...")
        return
    else:
        print("\nInvalid choice.")
        make_list()

# main method
todo = []

print("Do you want to create a todo list?")
make_list()

File: Todo.java
import java.util.Scanner;
import java.util.ArrayList;

public class Todo {
    // Class-level variable to hold the tasks
    static ArrayList<String> Task = new ArrayList<>();

    // Method to add a task
    static void addTask() {
        Scanner input = new Scanner(System.in);
        System.out.print("Task: ");
        String task = input.nextLine();
        Task.add(task);
        System.out.println("Task added.");
    }

    // Method to read tasks
    static void readTask() {
        if (Task.isEmpty()) {
            System.out.println("No tasks available.");
        } else {
            for (int i = 0; i < Task.size(); i++) {
                System.out.println((i+1) + ": " + Task.get(i));
            }
        }
    }

    // Method to update a task
    static void updateTask(int index) {
        if (index >= 0 && index < Task.size()) {
            Scanner input = new Scanner(System.in);
            System.out.print("Enter updated task: ");
            String newTask = input.nextLine();
            Task.set(index, newTask);
            System.out.println("Task updated.");
        } else {
            System.out.println("Invalid index.");
        }
    }

    // Method to delete a task
    static void deleteTask(int index) {
        if (index >= 0 && index < Task.size()) {
            Task.remove(index);
            System.out.println("Task deleted.");
        } else {
            System.out.println("Invalid index.");
        }
    }

    // Main method
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("\tWelcome Vishal!");
        System.out.println();

        int check;
        while (true) {
            System.out.println("Add\tRead\tUpdate\tDelete\tExit");
            System.out.println(" 1 \t 2 \t  3 \t  4 \t  5 ");
            System.out.println();

            System.out.print("Enter Command here - ");
            check = sc.nextInt();

            if (check == 1) {
                addTask();
            } else if (check == 2) {
                readTask();
            } else if (check == 3) {
                System.out.print("Index to update: ");
                int updateIndex = sc.nextInt();
                updateTask(updateIndex);
            } else if (check == 4) {
                System.out.print("Task to delete: ");
                int deleteIndex = sc.nextInt();
                deleteTask(deleteIndex);
            } else if (check == 5) {
                System.out.println("Exiting...");
                break;
            } else {
                System.out.println("Invalid command.");
            }
            System.out.println();
        }
        sc.close();
    }
}
